scripts/setup_ubuntu.sh

```diff
@@ -134,10 +138,10 @@ apt install -yq --allow-downgrades \
 
 # Toolchain variants
 if [[ -n "${SETUP_TOOLCHAIN_VARIANTS}" ]]; then
-  apt install -yq gcc-9 g++-9 clang-10
-  # in newer versions of ubuntu clang++-10 is installed as a part of clang-10
+  apt install -yq gcc-9 g++-9 clang-10 clang-12
+  # in newer versions of ubuntu clang++-NN is installed as a part of clang-NN
   if [[ "$VERSION" == "16" ]]; then
-    apt install clang++-10
+    apt install clang++-10 clang++-12
   fi
```

Comment:
No need to change this in this PR, but we don't support 16.04 any more (YAY!), so we can just start removing these cases when we encounter them.